[spellcheck] Part 6: Spell check directory pytest

shreyan-gupta opened the pull request.

wacban reviewed (approved):
Nice! 

It seems like csnake likes words to be separated like in snake_case or CamelCase - which I support. e.g. it doesn't like `levelname`. 

How do you decide between adding an inline `ignore` and adding it to the global exception list?

shreyan-gupta commented:
> Nice!
> 
> It seems like csnake likes words to be separated like in snake_case or CamelCase - which I support. e.g. it doesn't like `levelname`.
> 
> How do you decide between adding an inline `ignore` and adding it to the global exception list?

I figured there's an option like enable/disable compound words (link [here](https://cspell.org/configuration/document-settings/)) but didn't quite use it. That should be helpful in files that have a lot of compound words like `levelname`. For one offs, I just added it to the exception is it made sense like "levelname".

In terms of other words in general, it was totally upto my discretion. If the word made sense like "PGAS" and was used frequently enough across multiple files, then I just added it to the global exception list, else I added a one off exception. Examples of one off exceptions are some obscure python or rust library name.

shreyan-gupta commented:
I don't have a great standardization of when and when not to add to global exception list, but in case you have ideas, would love to enact on them!
